### PR TITLE
chore(tap): surface send error

### PR DESCRIPTION
**Pull request by pront**
A small improvement to error handling.

### CODE REVIEW

File: lib/vector-tap/src/lib.rs [hunk 1]
```diff
@@ -183,15 +183,19 @@ impl<'a> TapRunner<'a> {
...
-                                if sender_tx.send(output_events).await.is_err() {
-                                    debug!("Could not send events");
+                        if !output_events.is_empty() {
+                            match &self.output_channel {
+                                OutputChannel::Stdout(formatter) => {
+                                    self.output_event_stdout(&output_events, formatter);
+                                }
+                                OutputChannel::AsyncChannel(sender_tx) => {
+                                    if sender_tx.send(output_events).await.is_err() {
+                                        warn!("Could not send tap events");
```
Comment (pront) [inline lib/vector-tap/src/lib.rs:193]: Note to reviewers: Debatable here is we want to handle errors more granularly.

**Review by ArunPiduguDD — APPROVED** [review]

**Review by jszwedko — COMMENTED** [review]
It seems like an alternative here would be for whatever is consuming from this channel to be able to handle empty `Vec`s of events. Was that considered too and this deemed as preferable? I could see it being informative to the receiver to receive an empty set of events (that is it could take action based on that condition).

File: lib/vector-tap/src/lib.rs [hunk 2]
```diff
@@ -183,15 +183,19 @@ impl<'a> TapRunner<'a> {
...
+                                    }
```
Comment (bruceg) [inline lib/vector-tap/src/lib.rs:194]: Ideally this should also report the error, but it is just a test and the error is unlikely…
```suggestion
                                    if let Err(error) = sender_tx.send(output_events).await {
                                        warn!("Could not send tap events: {error}");
                                    }
```

**Review by bruceg — APPROVED** [review]

File: lib/vector-tap/src/lib.rs (hunk 2 above)
Comment (pront) [reply lib/vector-tap/src/lib.rs:194]: Good suggestion 👍

**Comment by pront** [comment]
> It seems like an alternative here would be for whatever is consuming from this channel to be able to handle empty `Vec`s of events. Was that considered too and this deemed as preferable? I could see it being informative to the receiver to receive an empty set of events (that is it could take action based on that condition).

This is a great point actually. An empty vector of events might be useful, e.g. it might indicates that all tap events were filtered out. I reverted this change. This PR is now just about surfacing a potential send error.

**Review by jszwedko — APPROVED** [review]

**Pull request merged**